ext/date/tests/DateTimeZone_compare_basic1.phpt

```diff
@@ -1,54 +1,60 @@
 --TEST--
-Test of compare object handler for DateTime objects
+Test of compare object handler for DateTimeZone objects
 --FILE--
 <?php
 
-// NB: DateTimeZone class does not define a customized compare class handler so standard object handler will be used
+$timezones = array(
+    ['+0200', '-0200'],
+    ['EST', 'PST'],
+    ['Europe/Amsterdam', 'Europe/Berlin']
+);
 
-echo "Simple test for DateTimeZone compare object handler\n";
-
-//Set the default time zone
-date_default_timezone_set("Europe/London");
-
-class DateTimeZoneExt1 extends DateTimeZone {
-}
-
-class DateTimeZoneExt2 extends DateTimeZone{
-	public $foo = "Hello";
-	private $bar = 99;
+foreach ($timezones as [$timezone1, $timezone2]) {
+    compare_timezones($timezone1, $timezone1);
+    compare_timezones($timezone1, $timezone2);
 }
 
-class DateTimeZoneExt3 extends DateTimeZoneExt2 {
+var_dump(new DateTimeZone('Europe/Berlin') == new DateTimeZone('CET'));
+
+function compare_timezones($timezone1, $timezone2)
+{
+    $tz1 = new DateTimeZone($timezone1);
+    $tz2 = new DateTimeZone($timezone2);
+    echo "compare $timezone1 with $timezone2\n";
+    echo "< ";
+    var_dump($tz1 < $tz2);
+    echo "= ";
+    var_dump($tz1 == $tz2);
+    echo "> ";
+    var_dump($tz1 > $tz2);
 }
 
-$obj1 = new DateTimeZone("Europe/London");
-$obj2 = new DateTimeZoneExt1("Europe/London");
-$obj3 = new DateTimeZoneExt2("Europe/London");
-$obj4 = new DateTimeZoneExt3("Europe/London");
-
-echo "\n-- All the following tests should compare equal --\n";
-var_dump($obj1 == $obj1);
-echo "\n-- All the following tests should compare NOT equal --\n";
-var_dump($obj1 == $obj2);
-var_dump($obj1 == $obj3);
-var_dump($obj1 == $obj4);
-var_dump($obj2 == $obj3);
-var_dump($obj2 == $obj4);
-var_dump($obj3 == $obj4);
-
 ?>
-===DONE===
---EXPECT--
-Simple test for DateTimeZone compare object handler
-
--- All the following tests should compare equal --
-bool(true)
-
--- All the following tests should compare NOT equal --
-bool(false)
-bool(false)
-bool(false)
-bool(false)
-bool(false)
+--EXPECTF--
+compare +0200 with +0200
+< bool(false)
+= bool(true)
+> bool(false)
+compare +0200 with -0200
+< bool(false)
+= bool(false)
+> bool(false)
```

Comment:
How can it both be not smaller, and not greater? :-)

Reply:
ZE only does `<` and `<=` comparisons. `>` and `>=` are mapped to the others by swapping operands. That means if you return 0 or 1 from a `compare_object` handler, but never -1, inequality comparisons will always yield `false`, i.e. they are not supported. This comes in quite handy wrt. timezones (if we ever will support inequality comparisons of timezones, we can support that in the handler).